Internal: Cover global classes analytics [ED-21201]

MuliElementor opened the pull request:
## PR Checklist
<!-- 
Please check if your PR fulfills the following requirements:
**Filling out the template is required.** Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
 -->
- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md


## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR using "x" with no spaces eg: [x]. -->
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

<!--start_gitstream_placeholder-->
### ✨ PR Description
```
Purpose: Add analytics tracking for global CSS classes to monitor user interactions with class management features.

Main changes:
- Added event tracking configuration for global CSS class interactions in eventsConfig.js
- Implemented tracking utility to capture class creation, deletion, renaming, and application events
- Added tracking calls to class management components and UI interactions for analytics collection
```

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using.
💡 **Tip:** You can customize your AI Description using **Guidelines** [Learn how](https://docs.gitstream.cm/automation-actions/#describe-changes)</sub>
<!--end_gitstream_placeholder-->

MuliElementor committed:
Enhance global classes tracking and event handling

- Added tracking for global class events including renaming, reordering, and usage interactions.
- Updated `onClearFilter` to accept a trigger type for better context in filter actions.
- Introduced new event tracking for search focus and filter interactions to improve analytics.
- Refactored components to ensure consistent event tracking across the global classes feature.

This update aims to improve user interaction insights and enhance the overall functionality of the global classes management system.

MuliElementor committed:
Enhance global class event tracking in Class Manager

- Added tracking for opening the class manager panel to capture user interactions.
- Updated the deleteClass function to include event tracking for class deletions, capturing class ID, title, and usage data.
- Modified the delete confirmation dialog to pass the class label when deleting a class.

These changes improve analytics and provide better insights into user actions within the global classes management system.

MuliElementor committed:
update

MuliElementor committed:
update imports

MuliElementor committed:
Refactor global class tracking and event handling

- Removed the deprecated `trackGlobalClassEvent` function and replaced it with `trackGlobalClasses` for consistent event tracking across the global classes feature.
- Updated various components to utilize the new tracking method for class creation, deletion, renaming, and filter interactions.
- Enhanced the `CssClassSelector` and `ClassManager` components to improve user interaction insights.

These changes streamline the tracking process and improve analytics for user actions within the global classes management system.

MuliElementor committed:
Enhance global classes tracking and configuration

- Introduced new global class tracking events for various user interactions, including class creation, deletion, and filter management.
- Updated the `events-config.js` to include a comprehensive set of global class-related CSS classes for better event handling.
- Refactored the `saveGlobalClasses` function to track publish conflicts more accurately.
- Added tracking for the opening of the class manager panel to improve user interaction insights.

These changes aim to streamline event tracking and enhance analytics for the global classes management system.

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Refactor tracking utilities to unify style tracking

- Replaced `trackGlobalClasses` with a new `trackStyles` function to streamline event tracking for CSS class interactions.
- Updated various components to utilize the new tracking function for events such as class creation, renaming, application, and state changes.
- Enhanced the tracking data structure to include additional context for class interactions, improving analytics capabilities.

These changes aim to improve the consistency and maintainability of tracking user interactions with CSS classes.

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Enhance tracking functionality in CSS class management

- Added location context to tracking events in StateMenuItem and UnapplyClassMenuItem for better analytics.
- Updated useCreateAction and useHandleSelect hooks to include location information in tracking events.
- Refactored trackStyles function to log provider status and event details for improved debugging.
- Removed unused tracking calls in useApply and useUnapply functions to streamline code.

These changes aim to improve the tracking of CSS class actions and enhance the overall analytics capabilities within the editor.

MuliElementor committed:
Enhance CSS class tracking and introduce prefetch functionality

- Updated UnapplyClassMenuItem to include provider information in tracking events for better analytics.
- Refactored trackStyles function to streamline tracking logic and improve debugging.
- Added PrefetchCssClassUsage component to prefetch CSS class usage data during initialization.
- Improved tracking event structure to ensure accurate data capture for class application and removal.

These changes aim to enhance the tracking capabilities and performance of CSS class management within the editor.

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Refactor tracking utility to improve analytics integration

- Moved the import of getMixpanel to the correct location for better organization.
- Simplified the track function by removing the testing parameter, streamlining the event tracking process.

These changes enhance the clarity and efficiency of the tracking utility within the editor's global classes.

MuliElementor committed:
fix tests

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
lint

MuliElementor committed:
lint

ronros-elementor commented on packages/packages/core/editor-global-classes/src/utils/tracking.ts:
```diff
@@ -0,0 +1,243 @@
+	};
+};
+
+const getRemovedInfo = async ( classId: StyleDefinitionID ) => {
```
should it be async?

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

ronros-elementor commented on packages/packages/core/editor-global-classes/src/utils/tracking.ts:
```diff
@@ -0,0 +1,243 @@
+	if ( ! cssClass.variants.length ) {
+		return { classTitle, totalStyleProperties: 0, hasCostumeCss: false };
+	}
+	const desktopView = cssClass.variants[ 0 ];
```
the first variant is not necessarily desktop, we may have a style definition with variants for tablet and/or mobile only

MuliElementor replied:
removed

MuliElementor committed:
update

MuliElementor committed:
Refactor global classes tracking and error handling

- Updated global classes configuration to use camelCase for class names.
- Introduced a new error type for tracking global classes events.
- Enhanced tracking function to accept a more structured data type.
- Adjusted components to align with the new tracking event structure.
- Fixed type definitions for tracking events to improve type safety.

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Refactor tracking event structure by removing unused properties

- Removed `totalStyleProperties` and `hasCustomCss` from `TrackingEventWithComputed` type and related functions.
- Simplified `extractCssClassData` to only return `classTitle`.
- Updated `getAppliedInfo` to reflect changes in the tracking event structure.

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
update

MuliElementor committed:
Refactor tracking module integration in tests

- Replaced direct tracking mocks with a centralized `mockTrackingModule` across various test files.
- Updated imports and mock implementations in `class-manager-button.test.tsx`, `class-manager-panel.test.tsx`, `global-classes-list.test.tsx`, `css-class-usage-popover.test.tsx`, `css-class-usage-trigger.test.tsx`, `active-filters.test.tsx`, `css-class-filter.test.tsx`, and `filter-list.test.tsx` to streamline tracking functionality in tests.

MuliElementor committed:
update

MuliElementor committed:
update

MuliElementor committed:
Refactor tracking event type definitions

- Removed unused event types and their associated structures from the tracking module.
- Updated the tracking function to utilize the refined event type mapping for improved type safety and clarity.

MuliElementor committed:
Remove unnecessary blank line in tracking utility file

MuliElementor committed:
lint

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics

MuliElementor committed:
Merge branch 'main' into ED-21201-cover-global-classes-analytics
